Gems/Terrain/Code/Source/TerrainRenderer/Components/TerrainSurfaceMaterialsListComponent.cpp

```diff
@@ -359,16 +345,29 @@ namespace Terrain
     }
 
     void TerrainSurfaceMaterialsListComponent::OnShapeChanged([[maybe_unused]] ShapeComponentNotifications::ShapeChangeReasons reasons)
-     {
-         AZ::Aabb oldAabb = m_cachedAabb;
+    {
+        AZ::Aabb oldAabb = m_cachedAabb;
 
-         LmbrCentral::ShapeComponentRequestsBus::EventResult(
-             m_cachedAabb, GetEntityId(), &LmbrCentral::ShapeComponentRequestsBus::Events::GetEncompassingAabb);
+        LmbrCentral::ShapeComponentRequestsBus::EventResult(
+            m_cachedAabb, GetEntityId(), &LmbrCentral::ShapeComponentRequestsBus::Events::GetEncompassingAabb);
 
-         TerrainAreaMaterialNotificationBus::Broadcast(
-             &TerrainAreaMaterialNotificationBus::Events::OnTerrainSurfaceMaterialMappingRegionChanged, GetEntityId(), oldAabb,
-             m_cachedAabb);
-     }
+        if (m_cachedAabb.IsValid() && !oldAabb.IsValid())
+        {
+            TerrainAreaMaterialNotificationBus::Broadcast(
+                &TerrainAreaMaterialNotificationBus::Events::OnTerrainSurfaceMaterialMappingRegionCreated, GetEntityId(), m_cachedAabb);
+        }
+        else if (!m_cachedAabb.IsValid() && oldAabb.IsValid())
+        {
+            TerrainAreaMaterialNotificationBus::Broadcast(
+                &TerrainAreaMaterialNotificationBus::Events::OnTerrainSurfaceMaterialMappingRegionDestroyed, GetEntityId(), oldAabb);
+        }
+        else
```

Comment:
With an else statement here it looks like OnTerrainSurfaceMaterialMappingRegionChanged will be called if the TerrainSurfaceMaterialsListComponent is activated and there is no shape component, because both the old aabb and the one retrieved from the ShapeComponentRequest bus would presumably be invalid. Is that desireable, given that OnTerrainSurfaceMaterialMappingRegionCreated would not have been called yet?

Reply:
I think you're right, if the new shape is invalid and the old shape is invalid then we just don't want to do anything. I'll change the else to `else if (m_cachedAabb.IsValid() && oldAabb.IsValid())`. Good catch.